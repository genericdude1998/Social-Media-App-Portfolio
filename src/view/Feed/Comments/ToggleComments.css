*{
    box-sizing: border-box;
    margin: 0px;
}

.baseOpenCommentsButton{
    border: 1px solid gray;
    padding: 2px;
    border-radius: 1px;
    margin-left: 5px;
}