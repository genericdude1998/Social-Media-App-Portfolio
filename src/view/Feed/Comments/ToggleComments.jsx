import React from 'react';
import styles from './ToggleComments.css';
import PropTypes from 'prop-types';

const ToggleComments = ({postId, openComments ,onOpenComments, onCloseComments}) => {
    return (
        !openComments ? 
            <button className={styles.baseOpenCommentsButton} onClick={() => onOpenComments(postId)}>Open Comments</button> 
            : 
            <button className={styles.baseOpenCommentsButton} onClick={() => onCloseComments(postId)}>Close Comments</button>
    );
}

ToggleComments.propTypes = {
    postId:PropTypes.number,
    openComments:PropTypes.bool, 
    onOpenComments:PropTypes.func,
    onCloseComments:PropTypes.func,

}

export default ToggleComments;
