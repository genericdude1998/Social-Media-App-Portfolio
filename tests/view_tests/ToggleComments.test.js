import React from "react";
import { shallow } from "enzyme";
import ToggleComments from "../../src/view/Feed/Comments/ToggleComments";
import {mockContent, mockId} from '../mockValues';

const mockOnCloseComments = jest.fn();
const mockOnOpenComments = jest.fn();

let wrapper =  shallow(
    <ToggleComments
        postId={mockId}
        openComments={true}
        onOpenComments={mockOnOpenComments}
        onCloseComments={mockOnCloseComments}
    />
)

describe('ToggleComments', () => {
    it('should match snap when given true open comments', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('should run onOpenComments with expected params when clicked', () => {
        wrapper.find('button').simulate('click');
        expect(mockOnCloseComments).toHaveBeenCalledWith(mockId)
    });
    it('should match snap when given false open comments', () => {
        wrapper.setProps({openComments:false})
        expect(wrapper).toMatchSnapshot();
    });
    it('should run onCloseComments with expected params when clicked', () => {
        wrapper.find('button').simulate('click');
        expect(mockOnOpenComments).toHaveBeenCalledWith(mockId)
    });
});