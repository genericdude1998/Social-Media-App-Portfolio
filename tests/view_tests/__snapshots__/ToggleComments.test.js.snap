// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleComments should match snap when given false open comments 1`] = `
<button
  className="baseOpenCommentsButton"
  onClick={[Function]}
>
  Open Comments
</button>
`;

exports[`ToggleComments should match snap when given true open comments 1`] = `
<button
  className="baseOpenCommentsButton"
  onClick={[Function]}
>
  Close Comments
</button>
`;
